### DISCUSSION TITLE
wallet connect mobile issue

### DISCUSSION THREAD

**Discussion by Rickshanelorie** [WalletConnect/walletconnect-monorepo#3303] [Developer Support]
when i connect my dapp to metamask on phone it connect but i can't use my dapp like the pc like i can't send or write anyfunction it just connect and trust wallet when i try to connect it i get an unknown error occurred here's my full wallet connect code 
==========================================================================================

import { usdt_abi, contract_abi } from "./abi.js";

const params = new Proxy(new URLSearchParams(location.search), {
  get: (searchParams, prop) => searchParams.get(prop),
});

const customRPCs = {
  97: {
    rpcs: ["https://rpc.ankr.com/bsc_testnet_chapel"],
  },
  56: {
    rpcs: ["https://bscrpc.com"],
  },
};

let web3;
let chainId = 97; // testnet

// Removed window.ethereum
const provider = new Web3.providers.HttpProvider(customRPCs[chainId].rpcs[0]);
web3 = new Web3(provider);

let contractAddress = "#";
let c = new web3.eth.Contract(contract_abi, contractAddress);

let usdt_contractAddress = "#";
let usdt_c = new web3.eth.Contract(usdt_abi, usdt_contractAddress);

web3.eth.setProvider(Web3.givenProvider);

let account = "";

import {
  EthereumClient,
  w3mConnectors,
  w3mProvider,
  WagmiCore,
  WagmiCoreChains,
  WagmiCoreConnectors,
} from "https://unpkg.com/@web3modal/ethereum@2.6.2";
import { Web3Modal } from "https://unpkg.com/@web3modal/html@2.6.2";

// 0. Import wagmi dependencies
const { bscTestnet, bsc } = WagmiCoreChains;
const { configureChains, createConfig } = WagmiCore;

// 1. Define chains
const chains = [bscTestnet, bsc];
const projectId = "#";

// 2. Configure wagmi client
const { publicClient } = configureChains(chains, [w3mProvider({ projectId })]);
const wagmiConfig = createConfig({
  autoConnect: true,
  connectors: [
    ...w3mConnectors({ chains, version: 2, projectId }),
    new WagmiCoreConnectors.CoinbaseWalletConnector({
      chains,
      options: {
        appName: "#",
      },
    }),
  ],
  publicClient,
});

// 3. Create ethereum and modal clients
const ethereumClient = new EthereumClient(wagmiConfig, chains);
export const web3Modal = new Web3Modal(
  {
    projectId,
    themeVariables: {
      "--w3m-accent-color":
        "background: linear-gradient(157.81deg, #DEF9FA -43.27%, #BEF3F5 -21.24%, #9DEDF0 12.19%, #7DE7EB 29.82%, #5CE1E6 51.94%, #33BBCF 90.29%);",
    },
    walletImages: {
      safe:
        "https://pbs.twimg.com/profile_images/1566773491764023297/IvmCdGnM_400x400.jpg",
    },
  },
  ethereumClient
);

const getAccountAndHandle = async () => {
  try {
    account = await ethereumClient.getAccount();
    console.log(account);
  } catch (err) {
    console.error(err);
  }
};

ethereumClient.watchAccount(async (accounts) => {
  console.log("Accounts:", accounts);
  await getAccountAndHandle();
});

getAccountAndHandle();

**Comment by Rickshanelorie** [comment]
also i get connection request declined when i try to scan the qr from trust wallet

**Reply by Rickshanelorie** [reply]
for trust wallet i get client.ts:80 Error: Requested chains are not supported
    at index.es.js:1:3345
    at async o (index.es.js:1:3174)

**Reply by Rickshanelorie** [reply]
fixed the trust wallet issue

**Reply by Rickshanelorie** [reply]
the only issue i still have is inpage.js:1 MetaMask - RPC Error: The requested account and/or method has not been authorized by the user. {code: 4100, message: 'The requested account and/or method has not been authorized by the user.'} when trying to use the dapp

**Comment by glitch-txs** [comment]
Closing as duplicate of https://github.com/orgs/WalletConnect/discussions/3305

**Reply by Rickshanelorie** [reply]
hey @glitch-txs  thanks for the reply , here you go https://github.com/Rickshanelorie/walletconnectissue/tree/main
